src/IO/GeoJSONWriter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Converter class from Geometry to GeoJSON.
+ */
+class GeoJSONWriter
+{
+    /**
+     * @param Geometry $geometry The geometry to export as GeoJSON.
+     *
+     * @return string The GeoJSON representation of the given geometry.
+     *
+     * @throws GeometryIOException If the given geometry cannot be exported as GeoJSON.
+     */
+    public function write(Geometry $geometry) : string
+    {
+        if ($geometry instanceof GeometryCollection
+            && ! is_subclass_of($geometry, GeometryCollection::class)) {
```

Comment:
Ah, I understand why now, because of your failing test earlier. This can be solved in a cleaner way by refactoring this method to perform the `instanceof GeometryCollection` check *after* all other checks, so that `MultiPoint`, `MultiLine` etc. are already handled.

Reply:
Actually it's fine as it is, now that I think about it: no other subclasses of GeometryCollection would make sense anyway, as GeoJSON is restricted in terms of allowed Geometries. Please just change this line to `&& $geometry->geometryType() === 'GeometryCollection'` and add an inline comment that this is to exclude MultiPoint, MultiLineString and MultiPolygon, for posterity.

Reply:
Resolved in commit: 99140e1